### PR TITLE
Delete Substrate Proposal Viewing

### LINKED ISSUE #5284: Web Links component not appearing on non-authored thread
## Describe the bug

The Web Links component on the thread page does not show up for threads that you are not the author of unless another component like polls or proposals already exists.

## To Reproduce

1. Create a thread with a web link.
2. In incognito load the same discussion page. The Web Links component will not render/show-up.
3. Now go back to your logged-in tab and create a poll.
4. Go back to the incognito window and refresh. The Web Links component should show up above the poll component.

## Expected behavior

Web Links component should render by itself if no other sidebar components are necessary.

## Additional context

A bug that wasn't found in #4829.

### LINKED ISSUE #5285: Substrate view thread page crashes when logged in with non-substrate account
## Describe the bug

The thread page crashes with the following error when viewing a Substrate thread (e.g. Edgeware) while logged in to an account that has not joined the community (no linked substrate account):
```
Cannot read properties of undefined (reading 'slice')
```

## To Reproduce

1. Login with a Metamask account that you know is not linked with any account that has joined the 'Edgeware' community.
2. Open any thread in the 'Edgeware' community.

## Expected behavior

Should be able to view the thread without any error.

## Additional context

The bug originates from [line 192][1] in `user.tsx` (`userAddress.slice`) because `userAddress` is undefined. It is undefined because an undefined `userAddress` is passed to the `User` component on [line 192][2] (lol what a coincidence) in `CreateComment.tsx`. The `userAddress` is undefined in the `CreateComment.tsx` component anytime the user has not joined the respective community but the prop type in `user.tsx` does not reflect this (i.e. should be optional like `userAddress?: string`). The page arises only for Substrate communities because the section where `.slice` is called on `userAddress` only executes when `app.chain.base === ChainBase.Substrate`.

[1]: https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/components/user/user.tsx#L192
[2]: https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/components/Comments/CreateComment.tsx#L192

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #4699
Closes: #5284

## Description of Changes
- Deletes all Substrate proposal pages, types, and controllers.
- Migration which removes Substrate chain-event subscriptions, notifications, and refactors proposal links into web links.

## "How We Fixed It"

- Fixed #5284 by updating the `showSidebar` component property in `ViewThreadPage.tsx`.
- Fixed #5285 by removing legacy Substrate identity user rendering in `user.tsx`

## Test Plan
- Login with Polkadot wallet
- Join a Substrate community i.e. 'edgeware'
- View and interact with a Substrate community by creating a thread, commenting, and reacting.
- View and click on the new web links by finding a post that previously had proposals linked (try ordering threads by like -> there should be a 'Pizza Party' thread that had proposals and now has web links).
- Now create your own thread and use web links to link any website.
- Ensure proposals/referenda/tips aren't options for Substrate communities in the left-hand sidebar.
- View Aave and Osmosis proposal pages to ensure they still function correctly.

## Deployment Plan

**WARNING: MUST DO STEP (1) BEFORE DEPLOYMENT SINCE ENV VAR IS USED IN MIGRATION**
1. `Set CE_DATABASE_URL` env var to the chain-events database URL
3. Deploy to CW
4. Delete the env var once the migration is complete since the database URL is subject to change over time.

## Other Considerations

- `Set CE_DATABASE_URL` needs to be set in the QA Heroku env in order for the migration to complete successfully. Failing to set the env var will _not_ cause the migration to fail but Substrate proposal links will not be updated and therefore cannot be tested.

### CODE REVIEW

**Comment by timolegros** [comment]
PR is ready for review but I'm leaving it in draft until #4829 is merged.

**Comment by timolegros** [comment]
Blocked by #5284.

**Comment by timolegros** [comment]
This PR now also needs updating because https://edgeware.subscan.io/ is broken (apparently Edgeware hasn't paid Subscan so Subscan is dropping them until they do) now so all the former proposal links and new proposal links created in the PR are invalid. Per @dillchen we should link out to https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fmainnet2.edgewa.re#/democracy. It is not possible to link specific proposals like https://edgeware.subscan.io/democracy_proposal/91 so all proposal links will point to the same place (albeit with different names/tags).

**Comment by CowMuon** [comment]
How far out does that put us on this one? Shaping up to be a super tight cycle/quarter

**Comment by dillchen** [comment]
It should be fine / not impact ship date, it’s a hardcoded URL / one page
that we redirect to, defer to Tim to confirm

I asked Tim to add this context here to give visibility to conversation

Sent via Superhuman iOS ***@***.***>

**Comment by timolegros** [comment]
> 2. I opened Pizza Party thread. I have seen linked proposals instead of web links and those links did not work. Is this because I did not run the migration beforehand?

It requires running the db script with the `CE_DATABASE_URL` env var defined like: `CE_DATABASE_URL=... yarn db-all`

I can send you the URL if you want to test it out.

**Review by jnaviask — APPROVED** [review]

File: packages/commonwealth/client/scripts/controllers/chain/ethereum/accounts.ts [hunk 1]
```diff
@@ -7,9 +7,7 @@ import type EthereumChain from './chain';
 
 // NOTE: this is just a boilerplate class; not verified to work yet.
 // TODO: hook this up to rest of the application and verify that it works
-class EthereumAccounts
-  implements IAccountsModule<EthereumCoin, EthereumAccount>
-{
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/controllers/chain/ethereum/accounts.ts:12]: Github actions -> 'EthereumCoin' is defined but never used

File: packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx [hunk 2]
```diff
@@ -344,7 +346,8 @@ const ViewThreadPage = ({ identifier }: ViewThreadPageProps) => {
           showLinkedProposalOptions ||
           showLinkedThreadOptions ||
           polls?.length > 0 ||
-          isAuthor
+          isAuthor ||
+          hasWebLinks
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx:350]: should we add `isAdmin` condition here as well? like we did below

File: packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx [hunk 3]
```diff
@@ -559,19 +562,21 @@ const ViewThreadPage = ({ identifier }: ViewThreadPageProps) => {
...
-                  <div className="cards-column">
-                    <LinkedUrlCard
-                      thread={thread}
-                      allowLinking={isAuthor || isAdminOrMod}
-                    />
-                  </div>
-                ),
-              },
-            ],
+            ...(isAuthor || isAdmin || hasWebLinks
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx:565]: should we add `isAdminOrMod` condition here? like we do below =>  `allowLinking={isAuthor || isAdminOrMod}`

File: packages/commonwealth/client/scripts/views/pages/proposals.tsx [hunk 4]
```diff
@@ -277,16 +213,12 @@ const ProposalsPage = () => {
...
-    (inactiveDemocracyProposals || [])
-      .map((proposal, i) => <ProposalCard key={i} proposal={proposal} />)
-      .concat(
-        inactiveCosmosProposals?.length
-          ? inactiveCosmosProposals.map((proposal) => (
-              <ProposalCard key={proposal.identifier} proposal={proposal} />
-            ))
-          : []
-      )
+    []
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/proposals.tsx:221]: How about => `inactiveCosmosProposals?.map((proposal) => (
  <ProposalCard key={proposal.identifier} proposal={proposal} />
)) || []` ?

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.tsx [hunk 5]
```diff
@@ -169,9 +165,7 @@ export const ThreadCard = ({
                   <CWTag
                     key={`${link.source}-${link.identifier}`}
                     type="proposal"
-                    label={`${chainEntityTypeToProposalShortName(
-                      'proposal' as IChainEntityKind
-                    )} 
+                    label={`Prop
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.tsx:168]: Just confirming if this should be `Proposal`?

File: packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx [hunk 6]
```diff
@@ -175,42 +132,7 @@ export const GovernanceSection = () => {
...
-  );
-
-  const matchesTipsRoute = matchRoutes(
-    [
-      { path: '/tips' },
-      { path: ':scope/tips' },
-      { path: `/proposal/${ProposalType.SubstrateTreasuryTip}` },
-      { path: `:scope/proposal/${ProposalType.SubstrateTreasuryTip}` },
-    ],
+    [{ path: '/proposals' }, { path: ':scope/proposals' }],
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx:135]: Not sure, but does this fail on `/Proposals`, `/pRoPosAls` or any other letter case combination?

Not a blocker but just for knowledge ^

File: packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx [hunk 7]
```diff
@@ -135,60 +134,27 @@ export const VoteListing = (props: VoteListingProps) => {
...
-                    </div>
-                  );
-              }
+              return (
+                <div className="vote" key={i}>
+                  <VoterInfo voter={vote.account} />
+                  <div className="vote-right-container">
+                    <CWText
+                      noWrap
+                      title={(vote as any).amount && (vote as any).amount}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx:143]: how about => `title={(vote as any)?.amount || ''}`

File: packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx [hunk 8]
```diff
@@ -135,60 +134,27 @@ export const VoteListing = (props: VoteListingProps) => {
...
+                    >
+                      {(vote as any).amount && (vote as any).amount}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx:145]: how about => `{(vote as any)?.amount || ''}`

**Comment by CowMuon** [comment]
Given we turned on automerge today, this automatically got merged when Jake submitted his approval. Malik added some questions here which should still be answered (obvi) @timolegros

File: packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx [hunk 9]
```diff
@@ -135,60 +134,27 @@ export const VoteListing = (props: VoteListingProps) => {
...
+                    </CWText>
+                    <CWText
+                      noWrap
+                      title={
+                        (vote as any).weight && `${(vote as any).weight}x`
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx:150]: => title={(vote as any)?.weight ? `${(vote as any).weight}x` : ''} 

thoughts?

File: packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx [hunk 10]
```diff
@@ -135,60 +134,27 @@ export const VoteListing = (props: VoteListingProps) => {
...
+                      }
+                    >
+                      {(vote as any).weight && `${(vote as any).weight}x`}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/proposals/vote_listing.tsx:153]: => `{(vote as any)?.weight ? ${(vote as any).weight}x : ''}`

thoughts?

File: packages/commonwealth/client/scripts/views/components/countdown.tsx [hunk 11]
```diff
@@ -33,9 +33,3 @@ export const Countdown = (props: CountdownProps) => {
 type CountdownUntilBlockProps = {
   block: number;
 };
-
-export const CountdownUntilBlock = (props: CountdownUntilBlockProps) => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/countdown.tsx:37]: Github actions => `'CountdownUntilBlockProps' is defined but never used`

File: packages/commonwealth/client/scripts/views/components/ProposalCard/ProposalCard.tsx [hunk 12]
```diff
@@ -90,21 +81,10 @@ export const ProposalCard = ({
       <div className="proposal-card-metadata">
         <div className="tag-row">
           <ProposalTag label={getPrimaryTagText(proposal)} />
-          {isNotNil(secondaryTagText) && (
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/ProposalCard/ProposalCard.tsx:93]: Github actions => `'isNotNil' is defined but never used`

**Review by mzparacha — APPROVED** [review]
A lot of changes, really nice work @timolegros. Added some minor suggestions.

LGTM 🚀

File: packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx (hunk 6 above)
Comment (timolegros) [reply packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx:135]: Works with any case.

**Comment by timolegros** [comment]
> A lot of changes, really nice work @timolegros. Added some minor suggestions.
> 
> LGTM 🚀

@mzparacha thanks for the review. Created https://github.com/hicommonwealth/commonwealth/issues/5351 to track the resolution of your comments in a follow-up PR.

**Pull request merged**